### ISSUE TITLE
Manjaro : last session not saved

### ISSUE THREAD

**Issue by Ste74** [linuxmint/slick-greeter#25]
Hi there , i have iported this greeter on Manjaro Linux but at the moment i have one issue : The session is not saved
Here the log of seat0
```
[+0,00s] DEBUG: slick-greeter.vala:511: Starting slick-greeter 1.0.0 UID=620 LANG=it_IT.UTF-8
Activating service name='org.a11y.atspi.Registry'
[+0,01s] DEBUG: settings.vala:143: Overriding dconf setting for background with /usr/share/backgrounds/manjarowall17_08.jpg
[+0,01s] DEBUG: settings.vala:143: Overriding dconf setting for theme-name with Arc-Maia-Dark
[+0,01s] DEBUG: settings.vala:143: Overriding dconf setting for icon-theme-name with Arc-Maia
[+0,01s] DEBUG: settings.vala:157: Overriding dconf setting for draw-grid
[+0,02s] DEBUG: slick-greeter.vala:517: Setting cursor
[+0,14s] DEBUG: slick-greeter.vala:531: Loading command line options
[+0,14s] DEBUG: slick-greeter.vala:559: Setting GTK+ settings
Successfully activated service 'org.a11y.atspi.Registry'
[+1,72s] DEBUG: slick-greeter.vala:582: Creating Slick Greeter
[+1,72s] DEBUG: slick-greeter.vala:55: Creating background surface
[+1,77s] DEBUG: Connecting to display manager...
[+1,77s] DEBUG: Wrote 26 bytes to daemon
[+1,77s] DEBUG: Read 8 bytes from daemon
[+1,77s] DEBUG: Read 160 bytes from daemon
[+1,77s] DEBUG: Connected api=1 version=1.22.0 default-session=default show-manual-login=false hide-users=false has-guest-account=false show-remote-login=true
[+1,86s] DEBUG: background.vala:457: Failed to load background component : Apertura del file «» non riuscita: File o directory non esistente
[+1,87s] DEBUG: background.vala:457: Failed to load background component : Apertura del file «» non riuscita: File o directory non esistente

(slick-greeter:609): Gtk-WARNING **: Attempting to add a widget with type GtkBox to a GtkMenuItem, but as a GtkBin subclass a GtkMenuItem can only contain one widget at a time; it already contains a widget of type GtkLabel
[+2,99s] DEBUG: Loading users from org.freedesktop.Accounts
[+2,99s] DEBUG: User /org/freedesktop/Accounts/User1001 added
[+2,99s] DEBUG: User /org/freedesktop/Accounts/User1002 added
[+2,99s] DEBUG: User /org/freedesktop/Accounts/User1000 added
[+3,00s] DEBUG: User /org/freedesktop/Accounts/User1003 added
[+3,00s] DEBUG: user-list.vala:1030: Adding/updating user stefano (Stefano)
[+3,00s] DEBUG: Loading sessions from org.freedesktop.DisplayManager
[+3,10s] DEBUG: slick-greeter.vala:193: Invalid session: 'gnome'. Using session 'mate' instead.
[+3,10s] DEBUG: user-list.vala:1030: Adding/updating user gnomeste ()
[+3,34s] DEBUG: slick-greeter.vala:193: Invalid session: 'gnome'. Using session 'mate' instead.
[+3,34s] DEBUG: user-list.vala:1030: Adding/updating user mateste ()
[+3,34s] DEBUG: slick-greeter.vala:193: Invalid session: 'mate'. Using session 'mate' instead.
[+3,34s] DEBUG: user-list.vala:1030: Adding/updating user stebudgie ()
[+3,35s] DEBUG: slick-greeter.vala:193: Invalid session: 'budgie-desktop'. Using session 'mate' instead.
[+3,40s] DEBUG: Loaded session /usr/share/xsessions/mate.desktop (MATE, Questa sessione esegue l'accesso in MATE)
[+3,44s] DEBUG: Loaded session /usr/share/xsessions/gnome-xorg.desktop (GNOME su Xorg, Questa sessione esegue l'accesso in GNOME)
[+3,45s] DEBUG: Loaded session /usr/share/xsessions/gnome.desktop (GNOME, Questa sessione esegue l'accesso in GNOME)
[+3,46s] DEBUG: Loaded session /usr/share/xsessions/gnome-classic.desktop (GNOME classico, Questa sessione si avvia con GNOME classico)
[+3,50s] DEBUG: Loaded session /usr/share/xsessions/budgie-desktop.desktop (Desktop Budgie, Questa sessione ti fa entrare nel Desktop Budgie)
[+3,52s] DEBUG: Loaded session /usr/share/wayland-sessions/gnome.desktop (GNOME, Questa sessione esegue l'accesso in GNOME)
[+3,53s] DEBUG: Starting authentication for user stefano...
[+3,53s] DEBUG: Wrote 23 bytes to daemon
[+3,54s] DEBUG: main-window.vala:198: Screen is 1366x768 pixels
[+3,54s] DEBUG: main-window.vala:206: Monitor 0 is 1366x768 pixels at 0,0
[+3,54s] DEBUG: slick-greeter.vala:585: Showing greeter
[+3,54s] DEBUG: slick-greeter.vala:266: Showing main window
[+3,55s] DEBUG: slick-greeter.vala:595: Starting main loop
[+3,56s] DEBUG: background.vala:485: Regenerating backgrounds
[+3,56s] DEBUG: background.vala:68: Making background /usr/share/backgrounds/manjarowall17_08.jpg at 1366x768
[+3,61s] DEBUG: Read 8 bytes from daemon
[+3,61s] DEBUG: Read 37 bytes from daemon
[+3,61s] DEBUG: Prompt user with 1 message(s)
[+3,67s] DEBUG: slick-greeter.vala:254: starting system-ready sound
[+3,85s] DEBUG: background.vala:121: Render of background /usr/share/backgrounds/manjarowall17_08.jpg complete
[+30,41s] DEBUG: session-list.vala:51: Adding session budgie-desktop (Desktop Budgie)
[+30,42s] DEBUG: session-list.vala:51: Adding session gnome (GNOME)
[+30,42s] DEBUG: session-list.vala:51: Adding session gnome (GNOME)
[+30,42s] DEBUG: session-list.vala:51: Adding session gnome-classic (GNOME classico)
[+30,42s] DEBUG: session-list.vala:51: Adding session gnome-xorg (GNOME su Xorg)
[+30,42s] DEBUG: session-list.vala:51: Adding session mate (MATE)
[+30,44s] DEBUG: greeter-list.vala:210: Error registering user list dbus object: Risulta già esportato un oggetto per l'interfaccia x.dm.SlickGreeter.List su /list
[+40,14s] DEBUG: Providing response to display manager
[+40,14s] DEBUG: Wrote 25 bytes to daemon
[+40,19s] DEBUG: Read 8 bytes from daemon
[+40,19s] DEBUG: Read 19 bytes from daemon
[+40,19s] DEBUG: Authentication complete for user stefano with return code 0
[+40,19s] CRITICAL: pango_layout_get_cursor_pos: assertion 'index >= 0 && index <= layout->length' failed
[+40,19s] DEBUG: Starting session gnome-xorg
[+40,19s] DEBUG: Wrote 22 bytes to daemon
[+40,55s] DEBUG: Read 8 bytes from daemon
[+40,55s] DEBUG: Read 4 bytes from daemon
[+40,55s] DEBUG: user-list.vala:1030: Adding/updating user stefano (Stefano)
[+40,56s] DEBUG: slick-greeter.vala:193: Invalid session: 'gnome-xorg'. Using session 'mate' instead.
[+40,56s] DEBUG: slick-greeter.vala:590: Got a SIGTERM
[+40,56s] DEBUG: slick-greeter.vala:598: Cleaning up
[+40,56s] DEBUG: slick-greeter.vala:606: AT-SPI exited with return value 0
[+40,56s] DEBUG: slick-greeter.vala:612: Exiting

```

**Comment by gabrc52** [comment]
It seems that the session validation does not work as intended, it checks for /usr/share/xsessions/yoursession instead of /usr/share/xsessions/yoursession.desktop

A temporary workaround would be to create the file it tries to check for, for example:
`sudo touch /usr/share/xsessions/gnome`
and do this for every session you have.

See https://github.com/linuxmint/slick-greeter/pull/28 (my attempt to fix the issue)

**Comment by Ste74** [comment]
Interesting . if upstream not adopt quickly i can do the workaround in manjaro
https://github.com/manjaro/packages-community/commit/4468f5676807a30908d9fe88343962e522393606

**Comment by Ste74** [comment]
@gabrc52 , merged your workaround , work like a charme :wink: thank you 
https://github.com/manjaro/packages-community/commit/afbcc1a2985da2e3b256d54befde02df5ecd6d50

**Comment by clefebvre** [comment]
Fix merged.

**Comment by snake-scaly** [comment]
I'm having a very similar issue with slick-greeter 1.2.2 in Mint 19.0. I'm trying to use i3 instead of Cinnamon. However the greeter always picks "Cinnamon (Default)" session regardless of how I logged in previously.

My ~/.dmrc lists "Session=i3" so this is correct. There are no errors in the seat0-greeter.log. There was /etc/lightdm/lightdm.conf.d/70-linuxmint.conf with "user-session=cinnamon" which I commented out. Still Cinnamon is chosen every time I log in. Is there a way to figure out what's wrong?
